app/src/main/java/com/android/bookswap/ui/profile/UserProfile.kt

```diff
@@ -0,0 +1,168 @@
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.sharp.ArrowBack
+import androidx.compose.material.icons.outlined.Edit
+import androidx.compose.material.icons.rounded.AccountCircle
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clipToBounds
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.android.bookswap.model.UserViewModel
+import com.android.bookswap.ui.theme.*
+import com.android.bookswap.user
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun UserProfile(userVM: UserViewModel = UserViewModel()) {
+  var user = userVM.getUser()
+  var showEditProfile by remember { mutableStateOf(false) }
+
+  var needRecompose by remember { mutableStateOf(false) }
+
+  if (showEditProfile) {
+    EditProfileDialog(
+        onDismiss = {
+          showEditProfile = false
+          needRecompose = true
+        },
+        onSave = {
+          userVM.updateUser(
+              greeting = it.greeting,
+              firstName = it.firstName,
+              lastName = it.lastName,
+              email = it.email,
+              phone = it.phoneNumber,
+              user.latitude,
+              user.longitude,
+              picURL = user.profilePictureUrl)
+          showEditProfile = false
+          needRecompose = true
+        },
+        dataUser = user,
+        modifier = Modifier.testTag("editProfileAld"))
+  }
+
+  LaunchedEffect(userVM.uuid, needRecompose) {
+    user = userVM.getUser()
+    needRecompose = false
+  }
+
+  BookSwapAppTheme() {
+
+    // Scaffold to provide basic UI structure with a top app bar
+    Scaffold(
+        modifier = Modifier.testTag("profileScreenContainer"),
+        topBar = {
+          TopAppBar(
+              modifier = Modifier.testTag("profileTopAppBar"),
+              // Title of the screen
+              title = { Text("Your Profile", modifier = Modifier.testTag("profileTitleTxt")) },
+              // Icon button for navigation (currently no action defined)
+              navigationIcon = {
+                IconButton(modifier = Modifier.size(32.dp), onClick = {}) {
+                  Icon(
+                      imageVector = Icons.AutoMirrored.Sharp.ArrowBack,
+                      contentDescription = "",
+                      modifier = Modifier.size(30.dp))
+                }
```

Comment:
Should use the already created back button.

Reply:
Not really relevant to this PR since the PR is for the data accesses using repository and source. It will be done in a future PR.